### PR TITLE
Fixed Unit and E2E test issues

### PR DESCRIPTION
**Pull request by tchow8**
# Description

[DIV-5354](https://tools.hmcts.net/jira/browse/DIV-5354)

Fixes # (issue)

## Type of change

Technical changes to remove excessive logs on unit tests and fix any issues with the tests.

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: middleware/helmet.js [hunk 1]
```diff
@@ -22,7 +23,7 @@ const setupHelmet = app => {
   const sha256s = [ config.ssl.hpkp.sha256s, config.ssl.hpkp.sha256sBackup ];
 
   // Helmet HTTP public key pinning
-  app.use(helmet.hpkp({ maxAge, sha256s }));
+  app.use(hpkp({ maxAge, sha256s }));
```
Comment (tchow8) [inline middleware/helmet.js]: helmet.hpkp is deprecated. The standalone hpkp module performs the same functionality.

File: package.json [hunk 2]
```diff
@@ -10,8 +10,8 @@
     "setup": "NODE_PATH=.",
     "dev": "NODE_PATH=. node-dev server.js",
     "mocks": "NODE_PATH=. node-dev mocks/index.js",
-    "test": "NODE_PATH=. NODE_ENV=testing LOG_LEVEL=ERROR nyc mocha --exit 'test/unit/**/*.test.js' --timeout 15000",
-    "test:unit": "NODE_PATH=. NODE_ENV=testing LOG_LEVEL=ERROR mocha --exit 'test/unit/**/*.js' --timeout 15000",
+    "test": "NODE_PATH=. NODE_ENV=testing LOG_LEVEL=OFF nyc mocha --exit 'test/unit/**/*.test.js' --timeout 15000",
```
Comment (tchow8) [inline package.json]: Turning LOG_LEVEL to off on unit tests. We have error test scenarios that will log the error message. For unit tests, we don't need to see any of these logs. Failing tests will be shown at the end of the test and in the report.

File: steps/co-respondent/cr-progress-bar/CrProgressBar.content.json [hunk 3]
```diff
@@ -63,7 +63,6 @@
     "improveHeading": "Help us improve this service",
     "improveContent": "Please help us improve this service by <a class=\"govuk-link\" href=\"http://www.smartsurvey.co.uk/s/8RR1T?pageurl=/done\" target=\"_blank\" aria-label=\"This link will open in a new tab.\">leaving your feedback</a>.",
     "help": "Getting help",
-    "yourCourt": "Your divorce centre",
```
Comment (tchow8) [inline steps/co-respondent/cr-progress-bar/CrProgressBar.content.json]: Duplicate key to L59

File: test/e2e/pages/co-respondent/crProgressBar.js [hunk 4]
```diff
@@ -43,11 +43,6 @@ function seePetitionToDownload() {
   I.see(content.en.files.dpetition);
 }
 
-function seeRespAnswersToDownload() {
```
Comment (tchow8) [inline test/e2e/pages/co-respondent/crProgressBar.js]: Co-Respondent is no longer allowed to view Respondent answers.

File: test/unit/middleware/helmet.test.js [hunk 5]
```diff
@@ -1,10 +1,13 @@
 const modulePath = 'middleware/helmet';
 
+const proxyquire = require('proxyquire');
 const helmet = require('helmet');
 const config = require('config');
-const setupHelmet = require(modulePath);
 const { sinon, expect } = require('@hmcts/one-per-page-test-suite');
 
+const hpkpStub = sinon.stub();
+const setupHelmet = proxyquire(modulePath, { hpkp: hpkpStub });
```
Comment (tchow8) [inline test/unit/middleware/helmet.test.js]: Proxyquire is used to easily stub modules which are also functions.

File: test/unit/services/idam.test.js [hunk 6]
```diff
@@ -131,7 +131,7 @@ describe(modulePath, () => {
     let sandbox = {};
 
     before(() => {
-      sandbox = sinon.sandbox.create();
+      sandbox = sinon.createSandbox();
```
Comment (tchow8) [inline test/unit/services/idam.test.js]: sandbox.create is the deprecated method of creating sandbox environments.

File: test/unit/steps/corespondent/crProgressBar.test.js [hunk 7]
```diff
@@ -264,7 +264,7 @@ describe(modulePath, () => {
     };
 
     it('shows correct content', () => {
-      content(CrProgressBar, basicSession, {
+      return content(CrProgressBar, basicSession, {
```
Comment (tchow8) [inline test/unit/steps/corespondent/crProgressBar.test.js]: Missing return meant the test was not returning the error to mocha, and even if the assertion failed, the test did not.

File: yarn.lock [hunk 8]
```diff
@@ -4245,7 +4245,7 @@ hosted-git-info@^2.1.4:
   resolved "https://registry.yarnpkg.com/hosted-git-info/-/hosted-git-info-2.7.1.tgz#97f236977bd6e125408930ff6de3eec6281ec047"
   integrity sha512-7T/BxH19zbcCTa8XkMlbK5lTo1WtgkFi3GvdWEyNuc4Vex7/9Dqbnpsf4JMydcfj9HCg4zUWFTL3Za6lapg5/w==
 
-hpkp@2.0.0:
+hpkp@2.0.0, hpkp@^2.0.0:
```
Comment (yarpo) [inline yarn.lock]: hpkp@2.0.0
hpkp@^2.0.0

they seem to be the same, aren't they?

**Pull request merged**